e2e/testcases/reconciler_manager_test.go

```diff
@@ -92,25 +92,37 @@ func TestManagingReconciler(t *testing.T) {
 
-	// test case 4: manually update the memory limit.
-	// The reconciler-manager manages the container resource requirements on both
-	// autopilot clusters and non-autopilot clusters. Any manual changes will be reverted.
-	rs := fake.RootSyncObjectV1Beta1(configsync.RootSyncName)
-	nt.T.Log("Manually update the first container CPU and Memory limit")
+	// test case 4:  the reconciler-manager should not revert the change to the fields that are not owned by reconciler-manager
+	nt.T.Log("Manually update fields that are not owned by reconciler-manager")
 	mustUpdateRootReconciler(nt, func(d *appsv1.Deployment) {
-		d.Spec.Template.Spec.Containers[0].Resources.Limits = corev1.ResourceList{
-			"cpu":    resource.MustParse(updatedFirstContainerCPULimit),
-			"memory": resource.MustParse(updatedFirstContainerMemoryLimit),
-		}
+		d.Spec.Template.Spec.Containers[0].TerminationMessagePath = "dev/termination-message"
+		d.Spec.Template.Spec.Containers[0].ImagePullPolicy = "Always"
+		d.Spec.Template.Spec.Containers[0].Stdin = true
+		d.Spec.Template.Spec.Tolerations = append(d.Spec.Template.Spec.Tolerations, corev1.Toleration{Key: "kubernetes.io/arch", Effect: "NoSchedule", Operator: "Exists"})
+		d.Spec.Template.Spec.PriorityClassName = "system-node-critical"
+		d.Spec.Paused = true
 	})
-	nt.T.Log("Verify the reconciler-manager should revert the memory limit change")
-	generation += 2 // generation bumped by 2 because the change will be first applied then reverted by the reconciler-manager
-	nomostest.Wait(nt.T, "the memory limit to be reverted", nt.DefaultWaitTimeout, func() error {
+
+	nt.T.Log("Verify the reconciler-manager does not revert")
+	generation++ // generation bumped by 1 because reconicler-manager should not revert this change
+	nomostest.Wait(nt.T, "the reconciler deployment to be updated", nt.DefaultWaitTimeout, func() error {
 		return nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem,
-			&appsv1.Deployment{}, hasGeneration(generation),
-			firstContainerMemoryLimitIs(managedMemoryLimit.String()))
+			&appsv1.Deployment{}, hasGeneration(generation), containerImagePullPolicy("Always"))
+	})
+	mustUpdateRootReconciler(nt, func(d *appsv1.Deployment) {
+		d.Spec.Template.Spec.Containers[0].TerminationMessagePath = "dev/termination-log"
+		d.Spec.Template.Spec.Containers[0].ImagePullPolicy = "IfNotPresent"
+		d.Spec.Template.Spec.Containers[0].Stdin = false
+		d.Spec.Template.Spec.Tolerations = originalTolerations
+		d.Spec.Template.Spec.PriorityClassName = ""
+		d.Spec.Paused = false
+	})
+	generation++ // generation bumped by 1 because reconciler-manager should not revert this change
+	nomostest.Wait(nt.T, "the reconciler deployment to be updated", nt.DefaultWaitTimeout, func() error {
+		return nt.Validate(nomostest.DefaultRootReconcilerName, v1.NSConfigManagementSystem,
+			&appsv1.Deployment{}, hasGeneration(generation), containerImagePullPolicy("IfNotPresent"))
```

Comment:
The test updates a few fields, but it only verifies the imagePullPolicy. It would be good to see other fields are verified as well.

Reply:
Done.